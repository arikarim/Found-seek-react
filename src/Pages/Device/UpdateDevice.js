import axios from "axios";
import React, { useEffect, useState } from "react";
import { Col, Container } from "react-bootstrap";
import { Redirect, useHistory } from "react-router";

const UpdateDevice = () => {
  const [device, setDevice] = useState("");
  const [uploadFile, setUploadFile] = useState([]);
  const [name, setName] = useState("");
  const [description, setDescription] = useState("");
  const [isChecked, setIsChecked] = useState(true);
  const history = useHistory();

  const handle = (e) => {
    const arr = Array.from(e.target.files);
    // console.log(arr);
    setUploadFile(arr);
  };
  const setDeviceData = async () => {
    try {
      const data = await axios.get("http://localhost:3001/devices");
      const id = history.location.pathname.slice(-1);
      const dev = data.data.filter((device) => device.id === Number(id));
      setDevice(dev[0]);
    } catch (e) {
      console.log(e);
    }
  };
  useEffect(() => {
    setDeviceData();
  });

  const submitForm = (event) => {
    event.preventDefault();
    let device = new FormData();
    device.append("name", name);
    device.append("description", description);
    device.append("availability", isChecked);

    uploadFile.map((e) => device.append("[images[]", e));
    const id = history.location.pathname;
    var thenum = id.replace(/^\D+/g, "");
    console.log(Number(thenum));
    axios
      .put(`http://localhost:3001/devices/${Number(thenum)}`, device, {
        headers: {
          "Content-Type": "multipart/form-data",
        },
      })
      .then((response) => {
        console.log(response);
        history.push("/my_devices");
        const alert = document.querySelector(".alert");
        alert.classList.remove("d-none");
        alert.classList.add("d-block");
        alert.innerHTML = "Device updated successfully";
        setTimeout(() => {
          alert.classList.add("d-none");
          alert.classList.remove("d-block");
        }, 3000);
        return <Redirect to={"/my_devices"} />;
      })
      .catch((error) => {
        history.push("/my_devices");
        const alert = document.querySelector(".alert");
        alert.classList.remove("d-none");
        alert.classList.add("d-block");
        alert.innerHTML = "Device updated successfully";
        setTimeout(() => {
          alert.classList.add("d-none");
          alert.classList.remove("d-block");
        }, 3000);
        console.log(error);
        return <Redirect to={"/my_devices"} />;
        // error response
      });
  };

  return (
    <Container>
      <Col md={6} className="mx-auto">
        <form
          className="form"
          onSubmit={submitForm}
          encType="multipart/form-data"
        >
          <label htmlFor="name">Name</label>
          <input
            onChange={(e) => setName(e.target.value)}
            className="form-control my-2"
            type="text"
            name="name"
            // value={name || device.name}
            id="name"
            placeholder={device && device.name}
            required
          />
          {/* {device && console.log(device)} */}
          <br />
          <label htmlFor="description">Description</label>
          <input
            onChange={(e) => setDescription(e.target.value)}
            className="form-control my-2"
            type="text"
            name="description"
            id="description"
            placeholder={device && device.description}
            required
          />
          <label htmlFor="file">Upload a file</label>
          <input
            name="file"
            className="form-control my-2"
            type="file"
            id="file"
            onChange={(e) => handle(e)}
            multiple
            required
          />
          <label htmlFor="availability">Available?</label>
          <input
            onChange={(e) => setIsChecked(e.target.checked)}
            className="mx-2"
            type="checkbox"
            name="availability"
            id="availability"
          />
          <br />
          <button className="btn btn-dark" type="submit">
            Update
          </button>
        </form>
      </Col>
    </Container>
  );
};

export default UpdateDevice;
