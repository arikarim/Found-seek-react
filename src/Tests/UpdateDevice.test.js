import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import UpdateDevice from "../Pages/Device/UpdateDevice";

describe("Pages", () => {
  test("should match with snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <UpdateDevice />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
