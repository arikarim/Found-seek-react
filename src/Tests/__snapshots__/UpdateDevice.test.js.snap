// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages should match with snapshot 1`] = `
<div
  className="height container"
>
  <div
    className="mx-auto col-md-6"
  >
    <form
      className="form"
      encType="multipart/form-data"
      onSubmit={[Function]}
    >
      <label
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form-control my-2"
        defaultValue=""
        id="name"
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <br />
      <label
        htmlFor="description"
      >
        Description
      </label>
      <input
        className="form-control my-2"
        defaultValue=""
        id="description"
        name="description"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <label
        htmlFor="file"
      >
        Upload a file
      </label>
      <input
        className="form-control my-2"
        id="file"
        multiple={true}
        name="file"
        onChange={[Function]}
        required={true}
        type="file"
      />
      <label
        htmlFor="availability"
      >
        Available?
      </label>
      <input
        className="mx-2"
        id="availability"
        name="availability"
        onChange={[Function]}
        type="checkbox"
      />
      <br />
      <button
        className="btn btn-dark"
        type="submit"
      >
        Update
      </button>
    </form>
  </div>
</div>
`;
